src/managers/balancer.py

```diff
@@ -18,6 +22,93 @@
 
 logger = logging.getLogger(__name__)
 
+RebalanceMode = Literal["full", "add", "remove"]
+
+
+class CruiseControlClient:
+    """Client wrapper for CruiseControl."""
+
+    def __init__(self, username: str, password: str, host: str = "localhost", port: int = 9090):
+        self.username = username
+        self.password = password
+        self.address = f"http://{host}:{port}/kafkacruisecontrol"
+        self.default_params = {"json": "True"}
+
+    def get(self, endpoint: str, **kwargs) -> requests.Response:
+        """CruiseControl GET request.
+
+        Args:
+            endpoint: the REST API endpoint to GET.
+                e.g `state`, `load`, `user_tasks`
+            **kwargs: any REST API query parameters provided by that endpoint
+        """
+        return requests.get(
+            f"{self.address}/{endpoint}",
+            # auth=(self.username, self.password),
+            params=kwargs | self.default_params,
+        )
+
+    def post(self, endpoint: str, dryrun: bool = False, **kwargs) -> requests.Response:
```

Comment:
**suggestion**: How about using a `Literal` with the accepted endpoints?

Reply:
I think right now it's not clear what endpoints we'll need yet! I don't want to encourage/limit at all, so just keeping it open-ended for now, as it'll have the same spec as the API itself.